rclpy/src/rclpy/utils.cpp

```diff
@@ -89,4 +97,103 @@ convert_to_py(void * message, py::object pyclass)
   return py::reinterpret_steal<py::object>(convert(message));
 }
 
+const char *
+get_rmw_implementation_identifier()
+{
+  return rmw_get_implementation_identifier();
+}
+
+void
+assert_liveliness(py::object pyentity)
+{
+  if (PyCapsule_IsValid(pyentity.ptr(), "rclpy_publisher_t")) {
+    auto publisher = static_cast<rclpy_publisher_t *>(
+      rclpy_handle_get_pointer_from_capsule(
+        pyentity.ptr(), "rclpy_publisher_t"));
+    if (RCL_RET_OK != rcl_publisher_assert_liveliness(&publisher->publisher)) {
+      throw RCLError("Failed to assert liveliness on the Publisher");
+    }
+  } else {
+    throw UnsupportedObjectTypeError("Passed capsule is not a valid Publisher.");
+  }
+}
+
+py::list
+remove_ros_args(py::object pycli_args)
+{
+  rcl_ret_t ret;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+
+  RCPPUTILS_SCOPE_EXIT(
+    {
+      ret = rcl_arguments_fini(&parsed_args);
+      if (RCL_RET_OK != ret) {
+        throw RCLError("Failed to fini parsed args");
+      }
+    });
```

Comment:
@gbalke consider setting this up _after_ parsing arguments (around line 164).

Reply:
My understanding is that you'd want this in place to clean up in case the parsing fails at any point (and throws). This will handle cleaning up the allocated memory? If I put it after then won't it not be set up and potentially leave allocated memory sitting around? Along the same lines I guess the `allocator` should also be in the same block.

Reply:
The reason for putting it after is [`rcl_arguments_fini()` will error if `rcl_arguments_t` hasn't been initialized](https://github.com/ros2/rcl/blob/b7784eb5d2fbff33053b69f6b9f9d3b703f9f2a7/rcl/src/rcl/arguments.c#L995), so this type should only be fini'd if initialization was successful.

`rcl_arguments_t` is the only type I know of that does this. I opened an issue to fix it in `rcl` https://github.com/ros2/rcl/issues/906